torch/_dynamo/variables/ctx_manager.py

```diff
@@ -1064,29 +1064,76 @@ def fn_name(self) -> str:
 
 class ProfilerContextVariable(ContextWrappingVariable):
     """
-    This class represents a set of torch profiler context objects, where Dynamo
-    ignores all the side-effects in the __init__, __enter__ and __exit__ methods
-    by treating the object mostly as a `contextlib.nullcontext`, except for edge
-    cases like the `__enter__` method which returns the object itself rather
-    than `None`, per implementation of the torch objects.
+    This class represents torch profiler context objects.
+
+    For record_function: emits torch.ops.profiler._record_function_enter_new
+    to the graph on enter, and torch.ops.profiler._record_function_exit on exit.
+    But if emit_profiler_ops=False, behaves like nullcontext.
+
+    For profile: behaves like nullcontext, ignoring all side-effects.
     """
 
-    def __init__(self, **kwargs: Any) -> None:
-        super().__init__(target_values=None, **kwargs)
+    _nonvar_fields = {
+        "emit_profiler_ops",
+        *ContextWrappingVariable._nonvar_fields,
+    }
+
+    @staticmethod
+    def create(
+        name: str,
+        args: Any = None,
+        emit_profiler_ops: bool = True,
+        **kwargs: Any,
+    ) -> "ProfilerContextVariable":
+        return ProfilerContextVariable(
+            target_values=[name, args] if emit_profiler_ops else None,
+            initial_values=None,
+            emit_profiler_ops=emit_profiler_ops,
+            **kwargs,
+        )
+
+    def __init__(
+        self,
+        target_values: Any = None,
+        initial_values: Optional[Any] = None,
+        emit_profiler_ops: bool = True,
```

Comment:
would be better to remove this flag and arg and just read from the config in the enter/exit, I don't see a case where this var would disagree with the config.

Reply:
remvoed